plugins/outputs/splunk/splunk.go

```diff
@@ -0,0 +1,277 @@
+package splunk
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers"
+)
+
+type Splunk struct {
+	Prefix string
+	Source string
+
+	SplunkUrl  string
+	AuthString string
+	Timeout    internal.Duration
+	client     *http.Client
+
+	SimpleFields    bool
+	MetricSeparator string
+	ConvertPaths    bool
+	ConvertBool     bool
+	UseRegex        bool
+	StringToNumber  map[string][]map[string]float64
```

Comment:
There's no documentation on this parameter, or what it's supposed to be doing.

Reply:
added comments throughout.